src/main/java/at/hannibal2/skyhanni/features/itemabilities/abilitycooldown/ItemAbilityCooldown.kt

```diff
@@ -4,272 +4,259 @@ import at.hannibal2.skyhanni.SkyHanniMod
-                }
-            }
-            // End Stone Sword
-            event.soundName == "mob.zombie.unfect" && event.pitch == 2.0f && event.volume == 0.3f -> {
-                ItemAbility.END_STONE_SWORD.sound()
-            }
-            // Soul Esoward
-            event.soundName == "mob.wolf.panting" && event.pitch == 1.3968254f && event.volume == 0.4f -> {
-                ItemAbility.SOUL_ESOWARD.sound()
-            }
-            // Pigman Sword
-            event.soundName == "mob.zombiepig.zpigangry" && event.pitch == 2.0f && event.volume == 0.3f -> {
-                ItemAbility.PIGMAN_SWORD.sound()
-            }
-            // Ember Rod
-            event.soundName == "mob.ghast.fireball" && event.pitch == 1.0f && event.volume == 0.3f -> {
-                ItemAbility.EMBER_ROD.sound()
-            }
-            // Fire Freeze Staff
-            event.soundName == "mob.guardian.elder.idle" && event.pitch == 2.0f && event.volume == 0.2f -> {
-                ItemAbility.FIRE_FREEZE_STAFF.sound()
-            }
-            // Staff of the Volcano
-            event.soundName == "random.explode" && event.pitch == 0.4920635f && event.volume == 0.5f -> {
-                ItemAbility.STAFF_OF_THE_VOLCANO.sound()
-            }
-            // Staff of the Volcano
-            event.soundName == "random.eat" && event.pitch == 1.0f && event.volume == 1.0f -> {
-                ItemAbility.STAFF_OF_THE_VOLCANO.sound()
-            }
-            // Holy Ice
-            event.soundName == "random.drink" && event.pitch.round(1) == 1.8f && event.volume == 1.0f -> {
-                ItemAbility.HOLY_ICE.sound()
-            }
+        for (ability in itemInHand.getActiveAbilities()) {
+            ability.onClick(event.clickType)
         }
     }
 
-    @SubscribeEvent
-    fun onItemClick(event: ItemClickEvent) {
-        if (AshfangFreezeCooldown.iscurrentlyFrozen()) return
-        handleItemClick(event.itemInHand)
-    }
+    private fun ItemStack.getActiveAbilities(): List<ActiveAbility> {
+        val internalName = getInternalName()
 
-    private fun handleItemClick(itemInHand: ItemStack?) {
-        if (!LorenzUtils.inSkyBlock) return
-        itemInHand?.getInternalName()?.run {
-            ItemAbility.getByInternalName(this)?.setItemClick()
+        val list = mutableListOf<ActiveAbility>()
+        for (ability in ItemAbilityType.entries) {
+            if (internalName in ability.internalNames) {
+                list.add(activeAbilities.getOrPut(ability) { ActiveAbility(ability) })
+            }
         }
+
+        return list
     }
 
     @SubscribeEvent
     fun onIslandChange(event: LorenzWorldChangeEvent) {
-        for (ability in ItemAbility.entries) {
-            ability.lastActivation = 0L
-            ability.specialColor = null
-        }
+        activeAbilities.clear()
     }
 
     @SubscribeEvent
     fun onActionBar(event: LorenzActionBarEvent) {
```

Comment:
Why not commented out complete? Does nothing right now.

Reply:
done